volplugin/utils.go

```diff
@@ -164,3 +189,26 @@ func splitPath(name string) (string, string, error) {
 func joinPath(tenant, name string) string {
 	return strings.Join([]string{tenant, name}, ".")
 }
+
+func addStopChan(name string) chan struct{} {
+	mountMutex.Lock()
+	stopChan := make(chan struct{}, 1)
+
+	if sc, ok := mountStopChans[name]; ok {
```

Comment:
Ideally, no. However, it's kind of protecting itself against bugs which might leak goroutines this way, so I thought it was appropriate.